homeassistant/components/weather/__init__.py

```diff
@@ -1088,6 +1111,51 @@ async def async_get_forecast_service(
     }
 
 
+async def async_get_forecast_value_service(
+    weather: WeatherEntity, service_call: ServiceCall
+) -> ServiceResponse:
+    """Get weather forecast for specified attribute."""
+    forecast_attribute = service_call.data.get("attribute")
+    forecast_limit_hours = service_call.data.get("limit_hours")
+    forecast_min_max = service_call.data.get("min_max")
+
+    supported_features = weather.supported_features or 0
+    forecast_type = "hourly"
+    if (supported_features & WeatherEntityFeature.FORECAST_HOURLY) == 0:
+        raise_unsupported_forecast(weather.entity_id, forecast_type)
```

Comment:
When can this happen? The service only allows calls on entities which supports forecasts?

Reply:
Since we call `weather.async_forecast_hourly()` we should check if the entity supports it
I looked at `def async_get_forecast_service`

Reply:
Right, but the `get_forecast` service can be called for an entity which supports hourly, twice daily or daily forecasts.
The service added by this can only be called if the entity supports hourly forecast so we don't need to check it again.

Reply:
Obsolete by implementing daily forecasts